### PR TITLE
Merge person

### PR DESCRIPTION
**Pull request by torhakon**

### CODE REVIEW

File: force-app/mergePerson/CRM_PersonMerge.cls [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+        accountIdList.addAll(this.accountsToMerge);
+
+        if (accountIdList.isEmpty() == false) {
+            for (User u : [
+                SELECT Id, AccountId, Account.INT_IsHasTombstone__c, IsActive
+                FROM User
+                WHERE AccountId IN :accountIdList AND IsActive = TRUE
+            ]) {
+                this.usersToRevokeAccess.add(u.Id);
+                if (u.IsActive) {
```
Comment (DeVaulted) [inline force-app/mergePerson/CRM_PersonMerge.cls:244]: This is always true, by definition of your search query.

File: force-app/mergePerson/CRM_PersonMerge.cls [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+        }
+        accountIdList.addAll(this.accountsToMerge);
+
+        if (accountIdList.isEmpty() == false) {
+            for (User u : [
+                SELECT Id, AccountId, Account.INT_IsHasTombstone__c, IsActive
+                FROM User
+                WHERE AccountId IN :accountIdList AND IsActive = TRUE
+            ]) {
+                this.usersToRevokeAccess.add(u.Id);
```
Comment (DeVaulted) [inline force-app/mergePerson/CRM_PersonMerge.cls:243]: This makes it seem like you're revoking access (=removing TPL) for all community users based on these accounts. Shouldn't there be a master community user left over? Or am I misunderstanding how the system works here?

File: force-app/mergePerson/CRM_PersonMerge.cls [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+                    this.userByAccountMap.put(u.AccountId, u);
+                }
+            }
+        }
+
+        return this;
+    }
+
+    /**
+     * @description Based on userByAccountMap, check if the
```
Comment (DeVaulted) [inline force-app/mergePerson/CRM_PersonMerge.cls:254]: Description cut off?

Comment (torhakon) [reply force-app/mergePerson/CRM_PersonMerge.cls]: Fixed

File: force-app/mergePerson/CRM_PersonMerge.cls (hunk 1 above)
Comment (torhakon) [reply force-app/mergePerson/CRM_PersonMerge.cls]: Fixed, removed the if clause

File: force-app/mergePerson/CRM_PersonMerge.cls (hunk 2 above)
Comment (torhakon) [reply force-app/mergePerson/CRM_PersonMerge.cls:243]: Yep, we remove all the TPLs if any.
If we only had one community user we could just let the TPL be. If there are two or more we would need to remove the one on the loosing side.
But to keep it simple we revoke all as they are regenerated when the user signs in again.

**Review by DeVaulted — APPROVED** [review]

**Pull request merged**